### ISSUE TITLE
bug: wrong constant infer

### ISSUE THREAD

**Issue by SolaWing** [castwide/solargraph#287]
```ruby
module A
  module AA
    class Gen
      def a
        pp a = A::Gen # infer to A::AA::Gen
        pp b = A::Gen::BB # can't infer
      end
    end
  end

  module Gen
    class BB
    end
  end
end
```

**Comment by castwide** [comment]
Confirmed. It's on the roadmap.

**Comment by castwide** [comment]
Fixed in the v0.39 branch.
